host/xtest/xtest_20000.c

```diff
@@ -0,0 +1,700 @@
+    TEEC_Session *sess, uint32_t obj, void *data, uint32_t data_size,
+    uint32_t *count)
+{
+    TEEC_Result res;
+    TEEC_Operation op;
+    uint32_t org;
+
+    memset(&op, 0, sizeof(op));
+    op.params[0].tmpref.buffer = data;
+    op.params[0].tmpref.size = data_size;
+    op.params[1].value.a = obj;
+    op.params[1].value.b = 0;
+
+    op.paramTypes = TEEC_PARAM_TYPES(
+        TEEC_MEMREF_TEMP_OUTPUT, TEEC_VALUE_INOUT, TEEC_NONE, TEEC_NONE);
+
+    res = TEEC_InvokeCommand(sess, TA_STORAGE_CMD_READ, &op, &org);
+
+    if (res == TEEC_SUCCESS)
+        *count = op.params[1].value.b;
+
+    return res;
+}
+
+static TEEC_Result obj_close(
+	TEEC_Session *sess, uint32_t obj)
+{
+	TEEC_Operation op;
+	uint32_t org;
+
+	memset(&op, 0, sizeof(op));
+	op.params[0].value.a = obj;
+
+	op.paramTypes = TEEC_PARAM_TYPES(
+		TEEC_VALUE_INPUT, TEEC_NONE, TEEC_NONE, TEEC_NONE);
+
+	return TEEC_InvokeCommand(sess, TA_STORAGE_CMD_CLOSE, &op, &org);
+}
+
+static int get_ta_dirname(TEEC_UUID *p_uuid, char *buffer)
+{
+
+	if (p_uuid == NULL || buffer == NULL)
+		return 0;
+	
+	return sprintf(buffer, "%08X%04X%04X%02X%02X%02X%02X%02X%02X%02X%02X",
+		       SWAP_BYTES_32(p_uuid->timeLow),
+		       SWAP_BYTES_16(p_uuid->timeMid),
+		       SWAP_BYTES_16(p_uuid->timeHiAndVersion),
+		       p_uuid->clockSeqAndNode[0],
+	               p_uuid->clockSeqAndNode[1],
+		       p_uuid->clockSeqAndNode[2],
+		       p_uuid->clockSeqAndNode[3],
+		       p_uuid->clockSeqAndNode[4],
+		       p_uuid->clockSeqAndNode[5],
+		       p_uuid->clockSeqAndNode[6],
+		       p_uuid->clockSeqAndNode[7]);
+}
+
+static int get_obj_filename(void *file_id, uint32_t file_id_length,
+			    char *buffer)
+{
+	char *p = buffer;
+	uint32_t i;
+
+	if (file_id == NULL || buffer == NULL)
+		return 0;
+
+	for (i=0; i<file_id_length; i++)
+		p += sprintf(p, "%02X", ((uint8_t *)file_id)[i]);
+
+	return p-buffer;
+}
+
+static int dump_file(FILE *fd, char *buffer, uint32_t len)
+{
+
+	uint16_t format = 16;
+	uint16_t size;
+	int res=-1;
+
+	printf("o Dump data (limit 128 bytes)\n");
+	if ( 0 != fseek(fd, 0, SEEK_SET))
+		goto exit;
+	fread(buffer, 1, len, fd);
+	size = ( len < 128 ) ? len : 128;
+	Do_ADBG_HexLog(buffer, size, format);
+	res=0;
+exit:
+	return res;
+}
+
+static int is_obj_present(TEEC_UUID *p_uuid, void *file_id,
+			  uint32_t file_id_length,
+			  enum tee_fs_file_type file_type,
+			  uint8_t block_num, uint8_t version)
+{
+	char ta_dirname[32 + 1];
+	char obj_filename[2*file_id_length + 1];
+	char cmd[150];
+
+	if (get_ta_dirname(p_uuid, ta_dirname) &&
+	    get_obj_filename(file_id, file_id_length, obj_filename)) {
+		if (file_type == META_FILE) {
+			sprintf(cmd,
+			"ls /data/tee/%s/%s/meta.%d >/dev/null 2>/dev/null",
+			ta_dirname, obj_filename, version);
+		}
+		if (file_type == BLOCK_FILE) {
+			sprintf(cmd,
+			"ls /data/tee/%s/%s/block%d.%d >/dev/null 2>/dev/null",
+			ta_dirname, obj_filename, block_num, version);
+		}
+		int res = system(cmd);
+		return res;
```

Comment:
This will return 0 when the file is present. It would be better to negate this value since the function name implies a boolean answer to "is object present".